pkgs/development/tools/ruff/default.nix

```diff
@@ -11,22 +11,16 @@
 
 rustPlatform.buildRustPackage rec {
   pname = "ruff";
-  version = "0.0.292";
+  version = "0.1.2";
 
   src = fetchFromGitHub {
     owner = "astral-sh";
-    repo = pname;
-    rev = "v${version}";
-    hash = "sha256-4D7p5ZMdyemDBaWcCO62bhuPPcIypegqP0YZeX+GJRQ=";
+    repo = "ruff";
+    rev = "refs/tags/v${version}";
```

Comment:
Why add this prefix?

Reply:
This is common practice now. Committers have asked me to do so in the past.
It is made to prevent confusion between same named but different tags and branches.

Reply:
I would love some comments on #257099, which should make this the default

Reply:
we can probably drop these type of prefixes treewide when https://github.com/NixOS/nixpkgs/pull/257099 is in.

Reply:
Sure, but do you think it will get merged anytime soon ?

Reply:
I meant this not as a blocker for this PR. But more as a post-cleanup for the other pr.